aoc/pkg/advent/advent.go

```diff
@@ -0,0 +1,108 @@
+package advent
+
+import (
+	"aoc/pkg/structs"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"os"
+	"strings"
+)
+
+func buildClient(urlObj *url.URL) (http.Client, error) {
+
+	sessionId := os.Getenv("AOC_SESSION_ID")
+	if sessionId == "" {
+		return http.Client{}, errors.New("no AOC_SESSION_ID provided. " +
+			"(It can be found by logging in through a browser and getting the session cookie.)",
+		)
+	}
+	jar, err := cookiejar.New(nil)
+	if err != nil {
+		return http.Client{}, err
```

Comment:
This would be more idiomatic `return nil, err` if you use pointer to http client.